data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAttesterDuties.java

```diff
@@ -106,47 +159,37 @@ public PostAttesterDuties(final DataProvider dataProvider, final JsonProvider js
         @OpenApiResponse(status = RES_SERVICE_UNAVAILABLE, description = SERVICE_UNAVAILABLE)
       })
   @Override
-  public void handle(Context ctx) throws Exception {
+  public void handle(@NotNull Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
     if (!validatorDataProvider.isStoreAvailable() || syncDataProvider.isSyncing()) {
-      ctx.status(SC_SERVICE_UNAVAILABLE);
+      AsyncApiResponse.respondWithError(
+          SC_SERVICE_UNAVAILABLE,
+          "Beacon node is currently syncing and not serving request on that endpoint");
       return;
     }
-    final Map<String, String> parameters = ctx.pathParamMap();
-    try {
-      final UInt64 epoch = UInt64.valueOf(parameters.get(EPOCH));
-      final UInt64[] indices = parseRequestBody(ctx.body(), UInt64[].class);
 
-      SafeFuture<Optional<PostAttesterDutiesResponse>> future =
-          validatorDataProvider.getAttesterDuties(
-              epoch, IntArrayList.toList(Arrays.stream(indices).mapToInt(UInt64::intValue)));
+    final UInt64 epoch = request.getPathParameter(EPOCH_PARAMETER);
+    final List<UInt64> indices = request.getRequestBody();
 
-      handleOptionalResult(
-          ctx, future, this::handleResult, this::handleError, SC_SERVICE_UNAVAILABLE);
+    SafeFuture<Optional<AttesterDuties>> future =
+        validatorDataProvider.getAttesterDuties(
+            epoch,
+            IntArrayList.toList(
+                Arrays.stream(indices.stream().mapToInt(UInt64::intValue).toArray())));
```

Comment:
tempting to try to clean this up...
if we need a list of integers, maybe we should just read that from the body rather than read uint64 and convert..